doc/release-notes.rst

```diff
@@ -30,6 +30,12 @@ Documentation
 * Documentation refactorization with examples. :issue:`472` by :user:`Guillaume
   Lemaitre <glemaitre>`.
 
+Maintenance
+...........
+
+* Remove :class:`pydicom.filereader.DicomIter` since it is unused. :issue:`482`
+  by :user:`Guillaume Lemaitre <glemaitre>`.
+
```

Comment:
Credit where credit is due, but does it make sense to add the author to selected issues in the release notes?